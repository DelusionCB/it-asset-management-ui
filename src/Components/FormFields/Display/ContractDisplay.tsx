import {Container} from 'reactstrap';
import React from 'react';
import {contractDataProps} from '../../../Types/types.directories';
import {DisplayDescription, DisplayHeader, DisplayText} from '../../CustomComponents/DisplayComponents';
import './index.scss';

interface contractDisplayProps {
    values: contractDataProps
    t: any
    navigate: any
}

function ContractDisplay ({values, t, navigate}: contractDisplayProps): JSX.Element {
    return (
        <Container className='directory-wrapper'>

            <DisplayHeader t={t} label='contract.con' />

            <DisplayDescription t={t} label='contract.description' />

            <DisplayText t={t} value={values.name} label='values.name' />
            <DisplayText t={t} value={values.description} label='values.description' />
            <DisplayText t={t} value={values.placeholder} label='values.placeholder' />

        </Container>
    )
}

export default ContractDisplay;
