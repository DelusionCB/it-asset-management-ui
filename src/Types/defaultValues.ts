import {
    contractDataProps, DirectoryResults,
    extendedApplicationDataProps,
    extendedDirectoryDataProps,
    LicenseDataProps, providerDataPropTypes,
    serverDataProps, serviceDataProps,
} from './types.directories';

export const appDefaultValues: extendedApplicationDataProps = {
    name: '',
    description: '',
    visibility: 'published',
    classification: 'tosi',
    customership: [],
    application_status: 'undefined',
    person_register: false,
    personal_info_logging: false,
    install_info: '',
    keywords: '',
    update_practice: '',
    security_practice_monitoring: '',
    recovery_practices_convalescence: '',
    log_archives: '',
    user_rights_management: '',
    security_solutions: '',
    product_owner: '',
    application_holder: '',
    admin_users: '',
    liability_professional_users: '',
    holder_extra_info: '',
    provider: null,
    provider_responsibility: '',
    additional_contacts: '',
    known_issues: '',
    contract: null,
    fileUrl: null,
    license: null,
    installed_server: [],
    service_dependency: [],
    integration: [],
    application_dependency: [],
}

export const licenseDefaultValues: LicenseDataProps = {
    base_id: '',
    id_prefix: '',
    name: '',
    description: '',
    created_time: '',
    last_modified_time: '',
    valid_from_date: '',
    valid_until_date: '',
    license_type: '',
    fileUrl: null,
    contract: '',
}

export const serverDefaultValues: serverDataProps = {
    name: '',
    description: '',
    base_id: '',
    id_prefix: '',
    applications: [],
    created_time: '',
    last_modified_time: '',
    visibility: '',
    server_role: '',
    service_level: '',
    operating_organization: '',
    place_of_use: '',
    product_owner: '',
    server_model: '',
    backup_data: '',
    backup_device: '',
    public_ip_addresses: '',
    dns_names: '',
    server_type: '',
    environment_type: '',
    dedicated: '',
    maintenance_window: '',
    device_criticality: '',
    security_level: '',
    status: '',
    install_date: '',
    ip_address: '',
    updates: '',
    verification_practices: '',
    recovery_practices_convalescence: '',
    logging: '',
    access_rights_management: '',
    security_solutions: '',
    external_rights: '',
    domain_name: '',
    sub_domain: '',
    ip_address_type: '',
    subnet_mask: '',
    default_gateway: '',
    mac_address: '',
}

export const contractDefaultValues: contractDataProps = {
    name: '',
    description: '',
    placeholder: '',
    visibility: 'published',
}

export const directoryDefaultValues: extendedDirectoryDataProps = {
    name: '',
    description: '',
    applications: [],
    servers: [],
    services: [],
    visibility: 'published',
}

export const providerDefaultValues: providerDataPropTypes = {
    name: '',
    description: '',
    visibility: 'published',
    business_id: '',
    provider_type: 'inhouse',
    full_address: '',
    switch_phone: '',
    general_email: '',
    support_phone: '',
    support_email: '',
    additional_contact: '',
    provider_user_contact: '',
    extra_url: '',
    related_contracts: [],
    related_applications: [],
    related_services: [],
}

export const serviceDefaultValues: serviceDataProps = {
    name: '',
    description: '',
    service_status: 'undefined',
    customership: [],
    criticality: 'normal',
    service_level: '',
    service_type: 'other',
    validity_type: 'ongoing',
    limitations: '',
    product_owner: '',
    service_holder: '',
    provider_role: '',
    provider_contact: '',
    additional_contacts: '',
    fileUrl: null,
    contract: null,
    provider: null,
    related_services: [],
    required_installations: [],
    visibility: 'published',
}

export const directoryDefaultResults: DirectoryResults = {
    count: 0,
    next: null,
    previous: null,
    results: [],
}
