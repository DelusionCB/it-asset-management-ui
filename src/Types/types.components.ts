
// eslint-disable-next-line import/named
import {NavigateFunction} from 'react-router-dom';
// eslint-disable-next-line import/named
import {TFunction} from 'i18next';
import {
    appDependencyItemProps,
    contractDataPropTypes, integrationDataPropTypes,
    licenseItemProps, providerDataPropTypes,
    serverDataPropTypes, serviceDataProps
} from "./types.directories";

// ItemGrid & Item

interface objectItem {
    name: string
    description: string
    id_prefix: string
    base_id: string
}

export interface itemPropTypes {
    itemKey: number
    item: objectItem
    t: TFunction
}

export interface gridPropTypes {
    items: objectItem[]
    t: TFunction
}

// Searchbar

export interface searchBarPropTypes {
    onFormSubmit: (searchQuery: string, contextType: string, fetch: boolean) => void
    t: TFunction
}

// ActionButton & ActionButtons

export interface actionButtonPropTypes {
    values: any
    action: string
    type: string
    mode: string
    disabled: boolean
}

export interface ActionPropTypes {
    values: any
    type: string
    mode: string
}
