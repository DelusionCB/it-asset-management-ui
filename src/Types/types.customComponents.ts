import React from 'react';
import {
    appDependencyItemProps,
    contractDataProps,
    licenseItemProps,
    serverDataProps,
    serviceDataProps,
} from './types.directories';

// eslint-disable-next-line import/named
import {NavigateFunction} from 'react-router-dom';
// eslint-disable-next-line import/named
import {TFunction} from 'i18next';


export interface textFieldProps {
    label: string
    validation?: object
    value?: string | undefined
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
    type: 'text' | 'file' | 'textarea'
    id: string
    placeholder: string
    disabled: boolean
    invalid: boolean
}

export interface visibilitySelectorProps {
    label: string
    validation?: object
    value: string | undefined
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
    options: string[]
    id: string
    placeholder: string
    disabled: boolean
    invalid: boolean
}

export interface customSelectionProps {
    onChange: (e: string) => void
    selections: string[]
    disabled: boolean
    label: string
}

export interface customSwitchProps {
    value: boolean
    label: string
    validation?: object
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
    id: string
    disabled: boolean
    invalid: boolean
}

export interface customSelectProps {
    label: string
    validation?: object
    value: object[] | object | undefined | null
    onChange: (value: MyOption | readonly MyOption[] | undefined | null, id: string) => void
    endpoint: string
    id: string
    placeholder: string
    disabled: boolean
    invalid: boolean
    isMulti: boolean
}

export interface MyOption {label: string, value: number}

// Display components

type ItemProps = serverDataProps | licenseItemProps | appDependencyItemProps | contractDataProps | serviceDataProps;

export interface basePropTypes {
    t: TFunction
    label: string
}

export interface objectPropTypes extends basePropTypes {
    value: ItemProps | null
    navigate: NavigateFunction
}

export interface arrayPropTypes extends basePropTypes {
    value: ItemProps[]
    navigate: NavigateFunction
}

export interface textPropTypes extends basePropTypes {
    value: string
}

export interface booleanPropTypes extends basePropTypes {
    value: boolean
}
