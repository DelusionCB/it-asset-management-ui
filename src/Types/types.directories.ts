
// Parameters used to modify fetch for directories
export interface DirectoryFetchParams {
    countNumber?: number | string
    pageNumber?: number | string
    endpoint?: string
}

export interface EndpointFetchParams {
    endpoint: string
    filter: string
}

// Parameters / props used to fetch items like applications or licenses
export interface itemFetchParams {
    itemId: string | undefined
    apiPath: string | undefined
}

// Types for results from fetch for directories
export interface DirectoryResults {
    count?: number
    next?: null | string
    previous?: null | string
    results: any[]
}

// Types for Directory context
export interface DirectoryContextType {
    directories: Record<string, DirectoryResults>
    loading: boolean
}

// Types / Props for Directory folder
export interface DirectoryFolderProps {
    data: any
    endpoints: string[]
}

// Types / Props for application data inside Directory data
export interface applicationDataProps {
    base_id: string
    id_prefix: string
    name: string
    description: string
}

export interface providerDataPropTypes {
    base_id?: string
    id_prefix?: string
    created_time?: string
    last_modified_time?: string
    name: string
    description: string
    visibility?: string
    business_id: string
    provider_type: string
    full_address: string
    switch_phone: string
    general_email: string
    support_phone: string
    support_email: string
    additional_contact: string
    provider_user_contact: string
    extra_url: string
    related_contracts: contractDataPropTypes[]
    related_applications: applicationDataProps[]
    related_services: serviceDataProps[]
}

// Types / Props for Directory data
export interface DirectoryDataProps {
    name: string
    description: string
    base_id: string
    applications: applicationDataProps[]
    created_time: string
    last_modified_time: string
    id_prefix: string
}

export interface serviceDataProps {
    base_id?: string
    base_token?: string
    id_prefix?: string
    name: string
    description: string
    created_time?: string
    last_modified_time?: string
    service_status: string
    customership: customerDataProps[]
    criticality: string
    service_level: string
    service_type: string
    validity_type: string
    limitations: string
    product_owner: string
    service_holder: string
    provider_role: string
    provider_contact: string
    additional_contacts: string
    fileUrl: string | null
    contract: contractDataPropTypes | null
    provider: providerDataPropTypes | null
    related_services: serviceDataProps[]
    required_installations: applicationDataProps[]
    visibility: string
}

export interface integrationDataPropTypes {
    base_id?: string
    base_token?: string
    id_prefix?: string
    name: string
    description: string
    created_time?: string
    last_modified_time?: string
    server_platform: serverDataPropTypes | null
    environment_type: string
    visibility: string
}

export interface extendedDirectoryDataProps {
    name: string
    description: string
    base_id?: string
    applications: applicationDataProps[]
    servers: serverDataPropTypes[]
    services: serviceDataProps[]
    created_time?: string
    last_modified_time?: string
    id_prefix?: string
    visibility: string
}

export interface contractDataPropTypes {
    base_id?: string
    created_time?: string
    last_modified_time?: string
    id_prefix?: string
    visibility: string
    name: string
    description: string
    contract_number: string
    contract_type: string
    provider_contact: string
    invoices_per_year: number
    value_per_year: number
    place_of_use: string
    valid_from_date: string
    valid_until_date: string
    contract_continuation: string
    contract_decisions: string
    contract_holder: string
    fileUrl: string | null
    provider: providerDataPropTypes | null
    related_applications: applicationDataProps[]
}

export interface licenseDataPropTypes {
    base_id?: string
    id_prefix?: string
    visibility: string
    name: string
    description: string
    audits: string
    created_time?: string
    last_modified_time?: string
    valid_from_date: string
    valid_until_date: string
    license_type: string
    fileUrl: string | null
    contract: contractDataPropTypes | null
}

export interface DirectoryItemProps {
    data: applicationDataProps
}

export interface serverDataPropTypes {
    name: string
    description: string
    base_id?: string
    id_prefix?: string
    applications: applicationDataProps[]
    created_time?: string
    last_modified_time?: string
    visibility: string
    server_role: string
    customership: customerDataProps[]
    place_of_use: string
    product_owner: string
    server_model: string
    backup_data: string
    backup_device: string
    public_ip_addresses: string
    dns_names: string
    server_type: string
    environment_type: string
    dedicated: string
    maintenance_window: string
    device_criticality: string
    security_level: string
    service_level: string
    server_status: string
    install_date: string
    ip_address: string
    updates: string
    verification_practices: string
    recovery_practices_convalescence: string
    logging: string
    access_rights_management: string
    security_solutions: string
    external_rights: string
    domain_name: string
    sub_domain: string
    ip_address_type: string
    subnet_mask: string
    default_gateway: string
    mac_address: string
}

export interface licenseItemProps {
    name: string
    description: string
    base_id: string
    id_prefix?: string
    valid_from_date: string
    valid_until_date: string
    license_type?: string
    contract?: string
}

export interface appDependencyItemProps {
    name: string
    description: string
    base_id: string
    id_prefix: string
}

// Types / Props for extended Application data
export interface extendedApplicationDataProps {
    base_id?: string
    name: string
    visibility: string
    description: string
    created_time?: string
    last_modified_time?: string
    classification: string
    customership: customerDataProps[]
    application_status: string
    person_register: boolean
    personal_info_logging: boolean
    install_info: string
    keywords: string
    update_practice: string
    security_practice_monitoring: string
    recovery_practices_convalescence: string
    log_archives: string
    user_rights_management: string
    security_solutions: string
    product_owner: string
    application_holder: string
    admin_users: string
    liability_professional_users: string
    holder_extra_info: string
    provider: providerDataPropTypes | null
    provider_responsibility: string
    additional_contacts: string
    known_issues: string
    contract: null
    fileUrl: string | null
    license: licenseItemProps | null
    installed_server: serverDataPropTypes[]
    service_dependency: ServiceDependencyDataProps[]
    integration: IntegrationsDataProps[]
    application_dependency: appDependencyItemProps[]
}

export interface ServiceDependencyDataProps {
    name: string
    description: string
    id_prefix: string
    base_id: string
}

export interface customerDataProps {
    name: string
    description: string
    id_prefix: string
    base_id: string
}

export interface IntegrationsDataProps {
    name: string
    description: string
    id_prefix: string
    base_id: string
}
