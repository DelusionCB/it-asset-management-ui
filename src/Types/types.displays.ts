import {
    contractDataPropTypes,
    extendedApplicationDataProps,
    extendedDirectoryDataProps, integrationDataPropTypes,
    licenseDataPropTypes, providerDataPropTypes, serverDataPropTypes, serviceDataProps,
} from './types.directories';
// eslint-disable-next-line import/named
import {NavigateFunction} from 'react-router-dom';
// eslint-disable-next-line import/named
import {TFunction} from 'i18next';

interface basePropTypes {
    t: TFunction
    navigate: NavigateFunction
}

export interface appDisplayPropTypes extends basePropTypes {
    values: extendedApplicationDataProps
}

export interface contractDisplayPropTypes extends basePropTypes {
    values: contractDataPropTypes
}

export interface directoryDisplayPropTypes extends basePropTypes {
    values: extendedDirectoryDataProps
}

export interface licenseDisplayPropTypes extends basePropTypes {
    values: licenseDataPropTypes
}

export interface serverDisplayPropTypes extends basePropTypes {
    values: serverDataPropTypes
}

export interface serviceDisplayPropTypes extends basePropTypes {
    values: serviceDataProps
}

export interface providerDisplayPropTypes extends basePropTypes {
    values: providerDataPropTypes
}

export interface integrationDisplayPropTypes extends basePropTypes {
    values: integrationDataPropTypes
}
