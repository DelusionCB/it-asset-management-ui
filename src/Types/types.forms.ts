
export interface formTypes {
    isDisabled: (is: boolean) => void
    type: string
    mode: string
    editValues: any
}
