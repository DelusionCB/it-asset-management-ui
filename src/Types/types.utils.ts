export type OptionalArray<T> = T[] | undefined | null;
export type OptionalObject<T> = Record<string, T> | undefined | null;
