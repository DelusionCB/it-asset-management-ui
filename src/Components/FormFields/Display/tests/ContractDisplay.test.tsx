import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ContractDisplay from '../ContractDisplay'
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import {contractDefaultValues} from '../../../../Types/defaultValues';

describe('<ContractDisplay />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <ContractDisplay t={jest.fn()} navigate={jest.fn()} values={contractDefaultValues} />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
