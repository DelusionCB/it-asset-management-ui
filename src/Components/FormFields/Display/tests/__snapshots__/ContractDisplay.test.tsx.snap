// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContractDisplay /> Snapshot renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <h1 />
  </div>
  <div
    className="row"
  >
    <hr
      hidden={true}
    />
    <h2 />
    <hr
      hidden={true}
    />
  </div>
  <div
    className="row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="row"
    >
      <h3 />
      <p />
    </div>
  </div>
</div>
`;
